src/index.tsx

```diff
@@ -23,22 +27,96 @@ async function main(GitURL: string, branchName?: string, folderPath?: string) {
     await $`git checkout ${branchName}`;
 
     await fs.remove(path.join(tempFolder, '.git'));
-    await fs.copy(
-        folderPath ? path.join(tempFolder, folderPath) : tempFolder,
-        targetFolder,
-        { overwrite: true }
-    );
+
+    const sourceFolder = folderPath
+        ? path.join(tempFolder, folderPath)
+        : tempFolder;
+
+    // Check if source is a file or directory
+    const sourceStat = await fs.stat(sourceFolder);
+
+    if (sourceStat.isFile()) {
+        // If it's a file, copy it directly to target directory with same name
+        const fileName = path.basename(sourceFolder);
+        await fs.copy(sourceFolder, path.join(targetFolder, fileName), {
+            overwrite: true
+        });
+    } else {
+        // If it's a directory, copy contents to target directory
+        await fs.copy(sourceFolder, targetFolder, { overwrite: true });
+    }
+}
+
+async function listSubmodules() {
+    const result = await $`git submodule status`;
+    console.log(`Current submodules:
+${result.stdout}
+Usage: xgit submodule remove <path>`);
+}
+
+async function removeSubmodule(submodulePath?: string) {
+    if (!submodulePath) {
+        await listSubmodules();
+        return;
+    }
+
+    console.log(`Removing submodule: ${submodulePath}`);
+
+    // Remove submodule entry from .gitmodules (if .gitmodules exists)
+    try {
+        await $`git config --file .gitmodules --remove-section submodule.${submodulePath}`;
+    } catch (error) {
+        console.warn(
+            `Warning: Could not remove from .gitmodules (file may not exist or section not found)`
+        );
+    }
```

Comment:
```suggestion
        await $`git config --file .gitmodules --remove-section submodule.${submodulePath}`;
```
不要吞掉命令行的原始报错！下面所有的都一样！